Framework/MDAlgorithms/src/IntegratePeaksMDHKL.cpp

```diff
@@ -0,0 +1,319 @@
+    }
+    double intensity = 0.0;
+    double errorSquared = 0.0;
+    integratePeak(neighborPts, histoBox, intensity, errorSquared);
+    p.setIntensity(intensity);
+    p.setSigmaIntensity(sqrt(errorSquared));
+    prog->report();
+    PARALLEL_END_INTERUPT_REGION
+  }
+  PARALLEL_CHECK_INTERUPT_REGION
+  // Save the output
+  setProperty("OutputWorkspace", peakWS);
+}
+
+MDHistoWorkspace_sptr IntegratePeaksMDHKL::normalize(
+    int h, int k, int l, double box, int gridPts, MatrixWorkspace_sptr flux,
+    MatrixWorkspace_sptr sa, IMDEventWorkspace_sptr ws) {
+  IAlgorithm_sptr normAlg = createChildAlgorithm("MDNormSCD");
+  normAlg->setProperty("InputWorkspace", ws);
+  normAlg->setProperty("AlignedDim0",
+                       "[H,0,0]," + boost::lexical_cast<std::string>(h - box) +
+                           "," + boost::lexical_cast<std::string>(h + box) +
+                           "," + boost::lexical_cast<std::string>(gridPts));
+  normAlg->setProperty("AlignedDim1",
+                       "[0,K,0]," + boost::lexical_cast<std::string>(k - box) +
+                           "," + boost::lexical_cast<std::string>(k + box) +
+                           "," + boost::lexical_cast<std::string>(gridPts));
+  normAlg->setProperty("AlignedDim2",
+                       "[0,0,L]," + boost::lexical_cast<std::string>(l - box) +
+                           "," + boost::lexical_cast<std::string>(l + box) +
+                           "," + boost::lexical_cast<std::string>(gridPts));
+  normAlg->setProperty("FluxWorkspace", flux);
+  normAlg->setProperty("SolidAngleWorkspace", sa);
+  normAlg->setProperty("OutputWorkspace", "mdout");
+  normAlg->setProperty("OutputNormalizationWorkspace", "mdnorm");
+  normAlg->executeAsChildAlg();
+  Workspace_sptr mdout = normAlg->getProperty("OutputWorkspace");
+  Workspace_sptr mdnorm = normAlg->getProperty("OutputNormalizationWorkspace");
+
+  IAlgorithm_sptr alg = createChildAlgorithm("DivideMD");
+  alg->setProperty("LHSWorkspace", mdout);
+  alg->setProperty("RHSWorkspace", mdnorm);
+  alg->setPropertyValue("OutputWorkspace", "out");
+  alg->execute();
+  IMDWorkspace_sptr out = alg->getProperty("OutputWorkspace");
+  return boost::dynamic_pointer_cast<MDHistoWorkspace>(out);
+}
+
+void IntegratePeaksMDHKL::integratePeak(const int neighborPts,
+                                        MDHistoWorkspace_sptr out,
+                                        double &intensity,
+                                        double &errorSquared) {
+  std::vector<int> gridPts;
+  const size_t dimensionality = out->getNumDims();
+  for (size_t i = 0; i < dimensionality; ++i) {
+    gridPts.push_back(static_cast<int>(out->getDimension(i)->getNBins()));
+  }
+
+  double *F = out->getSignalArray();
+  double Fmax = 0;
+  double Fmin = 1e300;
+  double sum = 0.0;
+  for (int i = 0; i < gridPts[0] * gridPts[1] * gridPts[2]; i++) {
+    if (!boost::math::isnan(F[i]) && !boost::math::isinf(F[i]) && F[i] != 0.0) {
```

Comment:
Is this check equivalent to [`std::is_normal(F[i])`](http://www.cplusplus.com/reference/cmath/isnormal/)?